packages/babel-helpers/src/helpers.js

```diff
@@ -461,6 +461,8 @@ helpers.set = template(`
 
       if (parent !== null) {
         set(parent, property, value, receiver);
+      } else {
+        receiver[property] = value;
```

Comment:
Reading spec is hard. So I just add another test case for accessors. This test case pass in V8, WebKit and Firefox so I assume it's correct.